src/svn_fs.ts

```diff
@@ -0,0 +1,262 @@
+import {
+  FileSystemProvider,
+  Event,
+  FileChangeEvent,
+  EventEmitter,
+  Uri,
+  Disposable,
+  FileStat,
+  FileType,
+  FileSystemError,
+  FileChangeType,
+  workspace
+} from "vscode";
+import * as path from "path";
+
+export class File implements FileStat {
+  type: FileType;
+  ctime: number;
+  mtime: number;
+  size: number;
+  name: string;
+  data?: Uint8Array;
+
+  constructor(name: string) {
+    this.type = FileType.File;
+    this.ctime = Date.now();
+    this.mtime = Date.now();
+    this.size = 0;
+    this.name = name;
+  }
+}
+
+export class Directory implements FileStat {
+  type: FileType;
+  ctime: number;
+  mtime: number;
+  size: number;
+
+  name: string;
+  entries: Map<string, File | Directory>;
+
+  constructor(name: string) {
+    this.type = FileType.Directory;
+    this.ctime = Date.now();
+    this.mtime = Date.now();
+    this.size = 0;
+    this.name = name;
+    this.entries = new Map();
+  }
+}
+
+export type Entry = File | Directory;
+
+class SvnFs implements FileSystemProvider, Disposable {
```

Comment:
Since this FS is in-memory and is not related to real svn repository I think it's worth renaming to InternalSvnFs or something like this, as well as scheme. Maybe it would useful to have virtual filesystem on top of real repository.

Reply:
what about `tempSvnFs` which is what it will be used for?

Reply:
Yes, I think it's good. And the scheme would look like for instance `tempsvnfs:/`.